actions/src/openstack_check_actions.py

```diff
@@ -0,0 +1,447 @@
+import json
+import os
+from typing import Callable
+import logging
+import sys
+from datetime import datetime
+import requests
+from openstack_api.openstack_connection import OpenstackConnection
+from openstack_action import OpenstackAction
+from requests.auth import HTTPBasicAuth
+
+
+class CheckActions(OpenstackAction):
+
+    # pylint: disable=useless-super-delegation
+    def __init__(self, *args, **kwargs):
+        """Constructor Class"""
+        super().__init__(*args, **kwargs)
+
+    # pylint: disable=arguments-differ
+    def run(self, submodule: str, **kwargs):
+        """
+        Dynamically dispatches to the method wanted
+        """
+        func: Callable = getattr(self, submodule)
+        return func(**kwargs)
+
+    # pylint: disable=too-many-arguments, too-many-locals, no-self-use
+    def check_project(
+        self, project: str, cloud: str, max_port: int, min_port: int, ip_prefix: str
+    ):
+        """
+        Does the logic for security_groups_check, should not be invoked seperately.
+        """
+        rules_with_issues = []
+
+        with OpenstackConnection(cloud_name=cloud) as conn:
+            sec_group = conn.list_security_groups(filters={"project_id": project})
+        # pylint: disable=too-many-nested-blocks
+        for group in sec_group:
+            for rules in group["security_group_rules"]:
+                # pylint: disable=line-too-long
+                if (
+                    rules["remote_ip_prefix"] == ip_prefix
+                    and rules["port_range_max"] == max_port
+                    and rules["port_range_min"] == min_port
+                ):
+                    ruledict = {}
+                    print("Issue with rule " + rules["security_group_id"])
+                    ruledict["id"] = rules["security_group_id"]
+                    ruledict["server_list"] = []
+
```

Comment:
Can we further chunk this down into sub methods e.g.:
```
def check_project():
   with OpenstackConnection(cloud_name=cloud) as conn:
            sec_group = conn.list_security_groups(filters={"project_id": project})
   bad_rules = self.filter_rules_with_issues(rules)
   rules_with_issues = self.find_rules_in_use(bad_rules)
```

Rather than it all being in a single method? A nice way method to tell your method does to much is describe the body without using the words `and` or `or`. At the moment I would say
"This checks projects by getting security groups and filters bad rules and those used in projects"

Instead it would be good to get it to
"This checks projects for invalid security rules" (then the child method do the filtering / checking)